From 5dfdfeade699244b5def096b6d740ec087bef8c5 Mon Sep 17 00:00:00 2001
From: Ravindra Reddy K <ravindra@marvell.com>
Date: Mon, 27 Nov 2017 14:09:00 +0530
Subject: [PATCH 2122/2241] fix: net: mv88e6xxx: remove MV88E6XXX_FLAG_PHY_ADDR
 flag for 88E6390 family

Unlike other devices where port numbers starts from 0x10, in 88E6390 family
port numbers starts from 0x0. When MV88E6XXX_FLAG_PHY_ADDR is set, port addr is
considered as 0x10 + port number. As a result when probing the switch, switch id
is read wrong and probe fails.
This issue fixed by removing MV88E6XXX_FLAG_PHY_ADDR flag for 88E6390 family
of devices.

Change-Id: I505f9adcfecceadc94549d5fb7aa63094e03549b
Signed-off-by: Ravindra Reddy K <ravindra@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/46885
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/47449
---
 drivers/net/dsa/mv88e6xxx.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/dsa/mv88e6xxx.c b/drivers/net/dsa/mv88e6xxx.c
index 0d6138d..eef6827 100644
--- a/drivers/net/dsa/mv88e6xxx.c
+++ b/drivers/net/dsa/mv88e6xxx.c
@@ -3641,7 +3641,7 @@ static const struct mv88e6xxx_info mv88e6xxx_table[] = {
 		.name = "Marvell 88E6390",
 		.num_databases = 4096,
 		.num_ports = 11,
-		.flags = MV88E6XXX_FLAGS_FAMILY_6390 | MV88E6XXX_FLAG_PHY_ADDR,
+		.flags = MV88E6XXX_FLAGS_FAMILY_6390,
 	},
 
 	[MV88E6290] = {
@@ -3650,7 +3650,7 @@ static const struct mv88e6xxx_info mv88e6xxx_table[] = {
 		.name = "Marvell 88E6290",
 		.num_databases = 4096,
 		.num_ports = 11,
-		.flags = MV88E6XXX_FLAGS_FAMILY_6390 | MV88E6XXX_FLAG_PHY_ADDR,
+		.flags = MV88E6XXX_FLAGS_FAMILY_6390,
 	},
 
 	[MV88E6190] = {
@@ -3659,7 +3659,7 @@ static const struct mv88e6xxx_info mv88e6xxx_table[] = {
 		.name = "Marvell 88E6190",
 		.num_databases = 4096,
 		.num_ports = 11,
-		.flags = MV88E6XXX_FLAGS_FAMILY_6390 | MV88E6XXX_FLAG_PHY_ADDR,
+		.flags = MV88E6XXX_FLAGS_FAMILY_6390,
 	},
 };
 
-- 
2.7.4

